### PR TITLE
Introduce Tooltip for the BlockClock navbar button

### PR DESCRIPTION
**Pull request by johnny9**
The Tooltip will appear when hovering over the BlockClock tab button in the desktop navigation bar. It will show the current state of IBD.

Two things are different than the figma. The wording for remaining blocks matches what is shown on the BlockClock instead of what the figma shows and After sync, Blockheight is shown instead. Looking for opinions on both during review.

vlcsnap-2024-05-01-20h05m06s361

[Build Artifacts](https://github.com/bitcoin-core/gui-qml/actions/runs/9012143092?pr=400)

### CODE REVIEW

File: src/Makefile.qt.include [hunk 1]
```diff
@@ -360,6 +361,7 @@ QML_RES_QML = \
   qml/components/StorageSettings.qml \
   qml/components/ThemeSettings.qml \
   qml/components/TotalBytesIndicator.qml \
+  qml/components/Tooltip.qml \
```
Comment (MarnixCroes) [inline src/Makefile.qt.include:364]: forgot to add this?

Comment (johnny9) [reply src/Makefile.qt.include:364]: whoops

**Comment by johnny9** [comment]
Update from dc31a22 to[ a7e4faa](https://github.com/bitcoin-core/gui-qml/commit/a7e4faa438e99d088d2773968613235cadb4c518)
 - Now with tooltip

File: src/Makefile.qt.include (hunk 1 above)
Comment (johnny9) [reply src/Makefile.qt.include:364]: thank you, updated the pr with the necessary files

**Review by MarnixCroes — COMMENTED** [review]
7453346a1ac518250a5600e3d37c4100c55feb8b
there should also be a tooltip state for when node is offline/stopped, i.e:
I pause the blockclock -> tooltip keeps displaying the blocktime from last online/sync time.
doesn't seem to be right

**Comment by johnny9** [comment]
> [7453346](https://github.com/bitcoin-core/gui-qml/commit/7453346a1ac518250a5600e3d37c4100c55feb8b) there should also be a tooltip state for when node is offline/stopped, i.e: I pause the blockclock -> tooltip keeps displaying the blocktime from last online/sync time. doesn't seem to be right

Good point. I will add some basic state status for paused/connected/offline

**Review by pablomartin4btc — COMMENTED** [review]
tACK 7453346a1ac518250a5600e3d37c4100c55feb8b

<details>
<summary>I agree with wording about time "<b>left</b>" matching the BlockClock instead of "<b>to go</b>" as in <a href="https://www.figma.com/file/ek8w3n3upbluw5UL2lGhRx/Bitcoin-Core-App-Design?type=design&node-id=8842-850575&mode=design">figma<a>, perhaps figma needs to be updated @GBKS?</summary>

- Figma:

  image

- This PR:

  image

</details>

It would be good to add the figma link and screenshot comparison in the description so reviewers could ACK easier. 

I also agree with @johnny9 and @MarnixCroes about adding different states wording as currently one can see the some inconsistencies:

<details>
<summary><b>Downloading blocks</b> (this one looks correct)</summary>

image

</details>

<details>
<summary><b>Paused</b></summary>

Screenshot from 2024-05-06 11-53-38

</details>

<details>
<summary><b>Connecting</b> (after pausing)</summary>

image

</details>

<details>
<summary><b>Connecting</b> (after sync completed/ blocktime determined)</summary>

image

</details>

<details>
<summary><b>Blocktime</b> determined (this one looks correct)</summary>

image

</details>

**Comment by johnny9** [comment]
Update  from 7453346 to bd62c7c 
 - Added paused and connecting stated
 
 
Screenshot from 2024-05-06 21-59-03
Screenshot from 2024-05-06 21-58-56
Screenshot from 2024-05-06 21-58-53

**Comment by johnny9** [comment]
Update from bd62c7c to 2478b9e 
 - newlline at end of file
 - Added copyright header to Tooltip.qml

**Comment by GBKS** [comment]
Good one. I tested and found a few things.

First, the screenshots:

image

As you can see, the text disappears in light mode, and the color of the tip does not adjust. On the web, it's possible to tie SVG colors to CSS color variables that automatically change when the theme changes. Not sure what the technique is for QML.

For me, it switched straight from "Downloading blocks - Estimating" to "Blocktime - xyz". It then counted blocks from ~1.5M or so to the current number. For much of that time it showed the almost full red circle, which felt like something was broken.

The tooltip should be placed 4-5px higher up, so the arrow tip overlaps with the button. From Figma:
image

Good call with the extra states. Can we list out which states we need? I started putting some of them together.
image

**Comment by GBKS** [comment]
Also a question. I see text directly in the code. To simplify localization, should those be text reference IDs instead so the actual copy can be grabbed from a locale-specific dictionary?

**Comment by johnny9** [comment]
> Also a question. I see text directly in the code. To simplify localization, should those be text reference IDs instead so the actual copy can be grabbed from a locale-specific dictionary?

the qsTr() wrapper function is how localization is managed with Qt

**Comment by johnny9** [comment]
> For me, it switched straight from "Downloading blocks - Estimating" to "Blocktime - xyz". It then counted blocks from ~1.5M or so to the current number. For much of that time it showed the almost full red circle, which felt like something was broken.

Unfortunately that is an old bug we're aware of. It has to do with how we determine if IBD is finished or not.

> Good call with the extra states. Can we list out which states we need? 

I think you got all of them. "Could not connect" is not included in my commit as we don't currently have that state until we integrate Network detection.

**Comment by johnny9** [comment]
Update to  to 8f8341f:
 - Fix tooltip text in lightmode
 - Adjusted the anchor upward 5 pixels so it aligns with the main component of the BlockClock mini button
 - Added light mode version of the tooltip arrow

**Review by D33r-Gee — COMMENTED** [review]
tACK 8f8341f6598afaa44e0370a67c7379fab78dd34c on WSL Ubuntu 22.04 LGTM

<details>
<summary>Screenshot</summary>

Screenshot 2024-05-14 110447


</details>

**Review by MarnixCroes — APPROVED** [review]
tack 8f8341f6598afaa44e0370a67c7379fab78dd34c

one q: why are `tooltip-arrow-dark.png` and `tooltip-arrow-light.png` different size?

**Review by pablomartin4btc — APPROVED** [review]
re-tACK 8f8341f6598afaa44e0370a67c7379fab78dd34c 

Checked all changes from 2478b9eda056f34fbdde46813f07f5141e0f2e3b as well.

**Pull request merged**